packages/vaadin-overlay/src/vaadin-overlay-position-mixin.js

```diff
@@ -87,18 +88,51 @@ export const PositionMixin = (superClass) =>
     constructor() {
       super();
 
-      this.__boundUpdatePosition = this._updatePosition.bind(this);
+      this._updatePosition = this._updatePosition.bind(this);
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+
+      if (this.opened) {
+        this.__addUpdatePositionEventListeners();
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      this.__removeUpdatePositionEventListeners();
+    }
+
+    /** @private */
+    __addUpdatePositionEventListeners() {
+      window.addEventListener('resize', this._updatePosition);
+
+      this.__positionTargetAncestorRootNodes = getAncestorRootNodes(this.positionTarget);
+      this.__positionTargetAncestorRootNodes.forEach((node) => {
+        node.addEventListener('scroll', this._updatePosition, true);
+      });
+    }
+
+    /** @private */
+    __removeUpdatePositionEventListeners() {
+      window.removeEventListener('resize', this._updatePosition);
+
+      if (this.__positionTargetAncestorRootNodes) {
+        this.__positionTargetAncestorRootNodes.forEach((node) => {
+          node.removeEventListener('scroll', this._updatePosition, true);
+        });
+        this.__positionTargetAncestorRootNodes = null;
+      }
     }
 
     __overlayOpenedChanged(opened) {
       // Toggle the event listeners that cause the overlay to update its position
-      ['scroll', 'resize'].forEach((eventName) => {
-        if (opened) {
-          window.addEventListener(eventName, this.__boundUpdatePosition);
-        } else {
-          window.removeEventListener(eventName, this.__boundUpdatePosition);
-        }
-      });
+      if (opened) {
+        this.__addUpdatePositionEventListeners();
```

Comment:
Seems that the positioning doesn't work when the overlay is opened for the first time. Used the date-picker dev page with padding added to the body for testing.

https://user-images.githubusercontent.com/1222264/170644804-413105db-cb6b-4f35-ab00-52c6b2967425.mp4

Reply:
Good catch. It is because the date-picker sets the position target only after the overlay is opened. I guess this will need adding a `positionTargetChanged` observer.

Reply:
Fixed. 781bc0255b1fc760b3cf478868c673f068a2ee40